### ISSUE TITLE
Unpracticle usage of ActiveForm::validate

### ISSUE THREAD

**Issue by tsvetann** [yiisoft/yii2#8318]
I just noticed that in the majority of situation where I would like to validate a form with ajax I return ActiveForm::validate(); But in any case I have the following code:

``` php
if ($model->load() && $model->save() {
....
end execution
}

return ActiveForm::validate($model);
```

That is just a rough example. I noticed that we perform a redundant second validation in ActiveForm::validate! Should we not check if $model already has the validation errors built from $model->save() and not run a validation second time or maybe have extra parameter to disable the second validation in ActiveForm::validate?

**Comment by cebe** [comment]
the code you show does not work as expected. If you want to do ajax validation, you should not save the model beforehand.

**Comment by tsvetann** [comment]
Thanks @cebe 

How do I validate and save via ajax then?

**Comment by cebe** [comment]
we could consider adding another parameter to `ActiveForm::validate()` to have full control over how model valiadate is called.

**Comment by tsvetann** [comment]
Thanks @cebe

**Comment by samdark** [comment]
Won't be implemented in 2.0.
